packages/components/src/autocomplete/autocompleter-ui.js

```diff
@@ -40,7 +48,16 @@ export function getAutoCompleterUI( autocompleter ) {
 			value,
 		} );
 
+		const [ needsA11yCompat, setNeedsA11yCompat ] = useState( false );
 		const popoverRef = useRef();
+		const popoverRefs = useMergeRefs( [
+			popoverRef,
+			useRefEffect( ( node ) => {
+				if ( node.ownerDocument !== contentRef.current.ownerDocument ) {
+					setNeedsA11yCompat( true );
+				}
```

Comment:
This may never realistically happen, but — should we also `setNeedsA11yCompat( false )` when the two `ownerDocument`s are the same?